src/GazeboThruster.cpp

```diff
@@ -15,9 +16,10 @@ GazeboThruster::~GazeboThruster()
 void GazeboThruster::Load(physics::ModelPtr _model,sdf::ElementPtr _sdf)
 {
     model = _model;
-    gzmsg << "GazeboThruster: loading thrusters from model: " << model->GetName() << std::endl;
+    gzmsg << "GazeboThruster: loading thrusters from model: " << model->GetName() << endl;
 
-    loadLinks();
+    loadThrusters();
```

Comment:
This is a typical situation where side-effects are harmful.

In loadThrusters, you fill the thrusters vector

In checkThrusters, you verify that the thrusters vector is valid.

This means that if you have an invalid thruster definition, you will end up with a class with an invalid state.

The better option would be:

~~~
std::vector<> thrusters = loadThrusters();
if (checkThrusters(thrusters))
   this->thrusters = thrusters;
~~~